concrete/src/Search/Pagination/PagerPagination.php

```diff
@@ -0,0 +1,164 @@
+<?php
+namespace Concrete\Core\Search\Pagination;
+
+use Concrete\Core\Search\ItemList\ItemList;
+use Concrete\Core\Search\ItemList\Pager\Manager\PagerManagerInterface;
+use Concrete\Core\Search\ItemList\Pager\PagerProviderInterface;
+use Concrete\Core\Search\Pagination\Adapter\PagerAdapter;
+use Concrete\Core\Search\Pagination\View\PagerManager;
+use Concrete\Core\Search\Pagination\View\PagerViewRenderer;
+use Concrete\Core\Search\Pagination\View\ViewRenderer;
+use Concrete\Core\Support\Facade\Facade;
+use Pagerfanta\Pagerfanta;
+
+class PagerPagination extends Pagination
+{
+
+    protected $list;
+    protected $app;
+    protected $request;
+    protected $cursor;
+    protected $hasNextPage;
+    protected $currentPageResults;
+
+    public function __construct(PagerProviderInterface $itemList)
+    {
+        $adapter = new PagerAdapter($itemList);
+        $this->list = $itemList;
+        $this->app = Facade::getFacadeApplication();
+
+        $manager = $itemList->getPagerManager();
+        $manager->sortListByCursor($itemList);
+
+        $factory = $itemList->getPagerVariableFactory();
+        $start = $factory->getCurrentCursor();
+        if ($start) {
+            $this->cursor = $start;
+            $manager->displaySegmentAtCursor($start, $itemList);
+        }
+
+        return Pagerfanta::__construct($adapter);
+    }
+
+    public function getLastResult()
+    {
+        return end($this->getCurrentPageResults());
+    }
+    
+    public function advanceToNextPage()
+    {
+        $results = $this->getCurrentPageResults();
+        $lastResult = end($results);
+
+        unset($this->currentPageResults);
+        unset($this->hasNextPage);
+
+        $manager = $this->list->getPagerManager();
+        $adapter = $this->getAdapter();
+        $this->cursor = $lastResult;
+        $manager->displaySegmentAtCursor($lastResult, $this->list);
+    }
+
+    public function renderView($driver = 'application', $arguments = array())
+    {
+        $manager = $this->app->make('manager/view/pagination/pager');
+        $driver = $manager->driver($driver);
+        $renderer = new ViewRenderer($this, $driver);
+        return $renderer->render($arguments);
+    }
+
+    public function getTotalPages()
+    {
+        return -1;
```

Comment:
I feel like this will break paging for everyone using the old style? is there any way to shim this? (maybe just if there is at-least 1 result return 2?